nodejs/week2/app.js

```diff
@@ -0,0 +1,88 @@
+const express = require("express");
+const app = express();
+const port = process.env.PORT || 3000;
+
+const document = require("./documents.json");
+
+// Support parsing JSON requests
+app.use(express.json());
+app.use(express.urlencoded({extended: true}));
+
+app.get("/", (req, res) => {
+  res.send("This is a search engine");
+});
+
+// GET /search
+app.get("/search", (req, res) => {
+    try {
+        const query = req.query.q;
+
+        if(!query) {
+            res.status(200).json(document);
+        } else {
+            const result = document.filter((obj) => 
+                Object.values(obj).some((value) => 
+                    String(value).toLowerCase().includes(query.toLowerCase())
+                )
+            );
+    
+            result.length === 0
+            ? res.sendStatus(404)
+            : res.status(200).json(result);
```

Comment:
Well, nice job @Nelia11 :)
I would suggest, only returning a json object, I mean the result will be enough rather then providing a 'Not Found' message.